File: problem/gcd/6/main.py

# main.py: main file for an "gcd" solution
#
# This is appended to the submitted Python program and tests the gcd function.
#
# The submitted file processes the standard input and this processes the
# "in.txt" file.
#
#
# Copyright (C) 2020  Rudy Matela

# standard input is already processed
# as the submitted programmed is run when imported
with open("in.txt") as filein:
  for line in filein:
    x,y = [int(x) for x in line.split()]
    print(gcd(x,y))

File: problem/gcd/6/main.js
// main.js: main file for solutions to the gcd exercise
//
// It processes values from standard input then from the "in.txt" file.
// This is appended to the submitted JavaScript program and tests the add function.
//
// The submitted file processes the standard input and this processes the
// "in.txt" file.
//
// Copyright (C) 2020-2021  Rudy Matela
// All rights reserved

var fs = require('fs').createReadStream('in.txt');
var rl = require('readline').createInterface({input: fs, output: process.stdout});

rl.on('line', function(line){
  words = line.split(/ /);
  a = parseInt(words[0]);
  b = parseInt(words[1]);
  console.log(gcd(a,b));
});
